File: OCH_Tao/ABC/B/224.py
H,W = map(int,input().split())
A = []
for i in range(H):
  A.append(list(map(int,input().split())))
for i1 in range(H-1):
  for i2 in range(i1+1,H):
    for j1 in range(W-1):
      for j2 in range(j1+1,W):
        if A[i1][j1]+A[i2][j2]>A[i2][j1]+A[i1][j2]:
          print("No")
          exit()
print("Yes")

File: OCH_Tao/ABC/B/224.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [H, W] = input.shift().split(" ").map(x => parseInt(x, 10));
let A = [];
for (const i of input) {
  A.push(i.split(" ").map(x => parseInt(x, 10)));
}
for (let i1 = 0; i1 < H - 1; i1++) {
  for (let i2 = i1 + 1; i2 < H; i2++) {
    for (let j1 = 0; j1 < W - 1; j1++) {
      for (let j2 = j1 + 1; j2 < W; j2++) {
        if (A[i1][j1] + A[i2][j2] > A[i2][j1] + A[i1][j2]) {
          console.log("No");
          process.exit();
        }
      }
    }
  }
}
console.log("Yes");